apps/prairielearn/src/middlewares/selectOnboardingTasksCounts.ts

```diff
@@ -0,0 +1,22 @@
+import asyncHandler from 'express-async-handler';
+
+import { getOnboardingSteps } from '../lib/onboarding.js';
+
+export default asyncHandler(async (req, res, next) => {
+  const course_id = res.locals.course?.id;
+
+  if (course_id && res.locals.course.show_onboarding) {
+    const steps = await getOnboardingSteps({
+      course_id,
+    });
```

Comment:
Nit, we can make this a little more concise. This is safe because `res.locals.course?.show_onboarding` won't ever evaluate to truthy if `res.locals.course` is undefined or null.

```suggestion
  if (res.locals.course?.show_onboarding) {
    const steps = await getOnboardingSteps({
      course_id: res.locals.course.id,
    });
```